process/block/preprocess/gasComputation.go

```diff
@@ -169,25 +170,28 @@ func (gc *gasComputation) ComputeGasConsumedByTx(
 		return 0, 0, process.ErrNilTransaction
 	}
 
-	txGasLimitConsumption := gc.economicsFee.ComputeGasLimit(txHandler)
+	moveBalanceConsumption := gc.economicsFee.ComputeGasLimit(txHandler)
 
 	txType := gc.txTypeHandler.ComputeTransactionType(txHandler)
-	isSCCall := txType == process.SCDeployment || txType == process.SCInvoking || txType == process.BuiltInFunctionCall
+	isSCCall := txType == process.SCDeployment ||
```

Comment:
I think that condition: txType == process.SCDeployment || txType == process.SCInvoking is not really necessary now as you have added now core.IsSmartContractAddress(txHandler.GetRcvAddr()) which covers these both too